exercises/practice/hello-world/.docs/instructions.md

```diff
@@ -8,7 +8,7 @@ or environment.
 
 The objectives are simple:
 
-- Write a function that returns the string "Hello, World!".
+- Define a variable containing the string "Hello, World!".
```

Comment:
the content of instructions.md must be exactly the same as those in https://github.com/exercism/problem-specifications/blob/main/exercises/hello-world/description.md. Therefore, this must be the following:

```suggestion
- Modify the provided code so that it produces the string "Hello, World!".
```

We have no choice in this matter.

Reply:
> the content of instructions.md must be exactly the same as those in https://github.com/exercism/problem-specifications/blob/main/exercises/hello-world/description.md. […]
> 
> We have no choice in this matter.

Might be worth mentioning that in a `CONTRIBUTING.md` or other place where new contributors are likely to look. Even though I tried to find and review this project's rules before filing this PR, I wasn't at all aware of this requirement.

As the current instructions (before this PR) already didn't match that specification, would it thus have been necessary to change them sooner or later, anyway?

> Therefore, this must be the following:
> ```suggestion
> - Modify the provided code so that it produces the string "Hello, World!".
> ```

Should we thus change the exercise and test to expect a function? Claiming that code with just a constant "produces" anything may not be completely wrong, but seems a bit far-fetched and not very intuitive.

Reply:
I don't think we would want to have the tests pass an argument that the function just ignores (since the function would in all likelihood just be `const "Hello, World!"`), so I don't think we should have `hello` be a function.

Reply:
1. There is something to be said for `hello` actually really producing something (laziness).
2. I do not expect confusion to arise from this word choice. Do you?

Reply:
> I actually think it's a rather strong one.

A strongly weakening factor is that a beginner is very unlikely to understand this argument, much less come up with it.

Reply:
> A strongly weakening factor is that a beginner is very unlikely to understand this argument, much less come up with it.

That may depend heavily on their previous knowledge of other programming languages and of other fields. Those not (yet) much aware of a compile-time vs. run-time distinction won't have much problems accepting it, I guess. Similarly, I imagine that to someone coming to programming from, say, linguistics (or even just theoretical computer science, where grammars _produce_ stuff), that argument might also seem quite natural. :shrug: